### PR TITLE
chore(argo-cd): Consolidated GPG configuration

### PR DESCRIPTION
**Pull request by pdrastil**
Closes: https://github.com/argoproj/argo-helm/issues/1471

- Merged GPG configuration keys together
- Added extra documentation how to use this
- Verified that GPG keys are working when imported in expected format

---

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/templates/NOTES.txt [hunk 1]
```diff
@@ -43,6 +43,12 @@ DEPRECATED option server.rbacConfig - Use configs.rbac
 {{- if .Values.configs.secret.argocdServerTlsConfig }}
 DEPRECATED option config.secret.argocdServerTlsConfig - Use server.certificate or server.certificateSecret
 {{- end }}
+{{- if or .Values.configs.gpgKeys }}
```
Comment (jmeridth) [inline charts/argo-cd/templates/NOTES.txt:46]: Did we mean to have the `or` here?

**Review by jmeridth — COMMENTED** [review]
LGTM just one question

File: charts/argo-cd/templates/NOTES.txt (hunk 1 above)
Comment (pdrastil) [reply charts/argo-cd/templates/NOTES.txt:46]: Yes, but annotation field is missing.

Comment (pdrastil) [reply charts/argo-cd/templates/NOTES.txt]: Fixed

**Review by jmeridth — APPROVED** [review]

**Comment by AllexVeldman** [comment]
Hi, I think this change broke the templating of the argocd-gpg-keys-cm ConfigMap.
https://github.com/argoproj/argo-helm/pull/1656.

**Comment by pdrastil** [comment]
Hi thanks for the fix

**Pull request merged**